include/boost/url/encode.hpp

```diff
@@ -208,17 +142,14 @@ encode(
         @ref encode_opts,
 */
 template<
-    class CharSet = grammar::all_chars_t,
-    class Allocator =
-        std::allocator<char> >
-std::basic_string<char,
-    std::char_traits<char>,
-        Allocator>
-encode_to_string(
+    BOOST_URL_STRTOK_TPARAM,
+    class CharSet = grammar::all_chars_t>
+BOOST_URL_STRTOK_RETURN
+encode(
     string_view s,
+    BOOST_URL_STRTOK_ARG(token),
     encode_opts const& opt = {},
-    CharSet const& allowed = {},
-    Allocator const& a = {});
+    CharSet const& allowed = {}) noexcept;
```

Comment:
umm... does encoding with `all_chars` as the allowed set make any sense? I have my doubts that these parameters are in the right order. I would think the charset comes first, since you cannot encode with out. Then the opts, and last the string token. We should discuss

Reply:
> does encoding with all_chars as the allowed set make any sense?

Regarding the `allowed` set, I think:

- the functionality is reasonable because only replacing all `%`s is something too simple to achieve without this function but
- the way we reached this parameter name led to something confusing because all characters are technically "allowed" in that encoding context. What we are really defining here is what chars are going to be encoded.

> I have my doubts that these parameters are in the right order. I would think the charset comes first, since you cannot encode with out. Then the opts, and last the string token. We should discuss

I've moved the token as the last parameter. 

Regarding the order of `opt` and `allowed`, I have no opinion. Considering that the `url` setters will reencode and especially when the default for `space_to_plus` is false, none of them are really mandatory.

In the meanwhile, I moved the charset before `opt` anyway.

Reply:
> the functionality is reasonable because only replacing all %s is something too simple to achieve without this function but

That's my point: a user never wants to "only encode percent characters" - this never comes up. Or do you have a motivating use-case?

> the way we reached this parameter name led to something confusing because all characters are technically "allowed" in that encoding context. What we are really defining here is what chars are going to be encoded.

"allowed" is short for "allowed unescaped".

We could rename it to "unreserved?" or "unescaped?" What do you think?

Reply:
> Or do you have a motivating use-case?

So the proposal is to not have a default parameter?

> We could rename it to "unreserved?" or "unescaped?" What do you think?

I think `unescaped` is the most descriptive

Reply:
Yes the proposal is not to have a default. If you like unescaped we can change it.